utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -278,7 +291,39 @@ pub(crate) fn new_year_in_sui<C: ChineseBased>(prior_solstice: RataDie) -> (Rata
     }
 }
 
-/// Get the moment of the nearest winter solstice on or before a given fixed date
+/// This function forces the RataDie to be on December 20, 21, 22, or 23. It was
+/// created for practical considerations and is not in the text.
+///
+/// See: <https://github.com/unicode-org/icu4x/pull/4904>
+fn bind_winter_solstice<C: ChineseBased>(solstice: RataDie) -> RataDie {
+    let (iso_year, iso_month, iso_day) = match iso_from_fixed(solstice) {
+        Ok(ymd) => ymd,
+        Err(_) => {
+            debug_assert!(false, "Solstice REALLY out of bounds: {solstice:?}");
+            return solstice;
+        }
+    };
+    if iso_month < 12 || iso_day < 20 {
+        #[cfg(feature = "logging")]
+        log::trace!("({}) Solstice out of bounds: {solstice:?}", C::DEBUG_NAME);
```

Comment:
nit: we should upgrade this to a debug assertion for iso_years that are ~±2000years.

Reply:
Done